Display revision comments in RevisionInformation.

nchevobbe opened the pull request:
Fixes Bug 1487761.

The revision comments were supposed to be displayed,
but we were looking at the wrong place to do so.
This patch fixes this, so revision comments are displayed
under the revision number + author.
The layout is changed to flex to do this properly.
Colors are tweaked a bit in order to pass WCAG standard.

This is how it looks:

<img width="1346" alt="image" src="https://user-images.githubusercontent.com/578107/70637859-3e77b400-1c38-11ea-82a6-65274926b530.png">

nchevobbe commented:
I don't understand the CI failure https://travis-ci.org/mozilla/treeherder/jobs/623740360?utm_medium=notification&utm_source=github_status

ionutgoldan commented:
Let me inspect that error locally.

ionutgoldan commented on ui/css/perf.css:
```diff
@@ -201,8 +201,18 @@ h1 {
+  list-style: none;
+  font-size: 0.7em;
+  font-style: italic;
+  color:  #757575;
```
You have an extra space here. This makes the CI linting test fail.

ionutgoldan commented:
![too-much-history](https://user-images.githubusercontent.com/12194756/70720600-c1624280-1cfc-11ea-96ec-81ce06d566df.png)

This is exposing all of the commit messages from a patch. That's not something we desire.
Could we show only the 1st message? And if there at least 2 messages, append a triple dot in the second list item?

nchevobbe commented:
I implemented the suggestion, and added the comments after the first one as a title on the `…` element:

<img width="1306" alt="image" src="https://user-images.githubusercontent.com/578107/70729591-c676c000-1d03-11ea-9c5c-af953f4b06f1.png">


Let me know what you think about that.
We can remove it, or put the `…` at the end of the first comment, and have the full comments as a title on the whole element.

I thought we could also play with `<details>` and `<summary>` but maybe that's an overkill

ionutgoldan commented:
Like your approach! 👍

ionutgoldan commented:
![uneven-widths](https://user-images.githubusercontent.com/12194756/70777457-860d5580-1d88-11ea-90b0-f0e9aa754044.png)

One last issue *(sorry for noticing it so late)*: the widths of the 2 list items *(for Base and New)* aren't equal, especially if the displayed commit message is bigger. We need to ensure they are the same & rather wrap the text of the message.

nchevobbe commented:
> One last issue (sorry for noticing it so late): the widths of the 2 list items (for Base and New) aren't equal, especially if the displayed commit message is bigger. We need to ensure they are & rather wrap the text of the message.
> Basically, ensure the — between Base and New is always centered in the middle of the page's width.

That should be okay now.

ionutgoldan commented:
![broken-aggregated-baseline](https://user-images.githubusercontent.com/12194756/70792242-764e3b00-1da1-11ea-80a4-c267eeac46b9.png)

Hmm... Looks like our changes break the aggregated baselines *(when we don't select a specific revision for the baseline)*. I need to read the code changes thoroughly.

nchevobbe commented:
> Hmm... Looks like our changes break the aggregated baselines (when we don't select a specific revision for the baseline). I need to read the code changes thoroughly.

oh, I should have check, sorry.

ionutgoldan commented:
Looking good now! Thanks!

ionutgoldan commented:
FYI @karlht.

ionutgoldan commented:
Nicholas, you'll just have to wait a bit more for Treeherder's final code review.

nchevobbe commented:
anything needed from me?

octavian-negru commented:
I don't have the rights to merge your work.
@ionutgoldan have merge rights but is on PTO until 3 January.
@camd do you have the rights to merge this work?

nchevobbe commented:
Hello there! Can this be merged? Unless there's some conflict? Thanks!

ionutgoldan commented:
Hi, Nicolas! Could you please rebase this PR on latest master? Thanks!
I no longer have merge rights, but will further your request.

nchevobbe commented:
> Hi, Nicolas! Could you please rebase this PR on latest master? Thanks! 

DONE :)

>  I no longer have merge rights, but will further your request.

Thanks!
